src/v/datalake/partition_key_path.cc

```diff
@@ -305,7 +297,7 @@ checked<remote_path, partition_key_error> partition_key_to_path(
             return element_key.error();
         }
         auto element = ssx::sformat(
-          "{}={}", element_key.value(), formatting_res.value());
+          "{}%3D{}", element_key.value(), formatting_res.value());
```

Comment:
Doesn't this result in a double encoding problem if we encode equal sign here and then encode `%` again in make requests (url_encode_target)?

Reply:
no, this is really confusing, i got the problem of understating that as well. The double encoding is actually required here as we want the encoded path to be received by Object Store as it requires special characters to be encoded. Not for the API but for the storage itself. 
So we want object name to contain `%3D` and in this case the `%` must be encoded when sending requests over HTTP protocol

Reply:
I tried running the test you just added but modified it to sleep before exiting. Next I used minio client to list the keys:

```
bash-5.1# mc ls myminio/panda-bucket-b757fa9c-f8e7-11ef-a61f-0242c0a8d713/redpanda-iceberg-catalog/redpanda/test/
[2025-03-04 11:00:14 UTC]     0B metadata/
[2025-03-04 11:00:14 UTC]     0B redpanda.timestamp_hour%3D2025-03-04-10/
```

I have also used PyIceberg before with minio and here are what it's paths look like:

```
bash-5.1# mc ls myminio/localhost/warehouse/demo_table/data
[2025-02-25 11:31:57 UTC]     0B string=HjvjcmahNe/
[2025-02-25 11:31:57 UTC]     0B string=JkKbb5ZW6D/
[2025-02-25 11:31:57 UTC]     0B string=LcCjv0beCR/
```

Our path format doesn't match PyIceberg but I believe they should. Do you think otherwise?

Reply:
i read the naming doc once again and i think that we can use `=` sign instead of `%3D`

Reply:
i will change the ordering as well